superset/datasets/commands/exceptions.py

```diff
@@ -173,6 +173,10 @@ class DatasetRefreshFailedError(UpdateFailedError):
     message = _("Dataset could not be updated.")
 
 
+class DatasetSamplesFailedError(CommandInvalidError):
+    message = _("Dataset could not be sampled.")
```

Comment:
nit/suggestion - this might be more explicit:
```suggestion
class DatasetSamplesFailedError(CommandInvalidError):
    message = _("Samples for dataset could not be retrieved.")
```